internal/policy/policy_test.go

```diff
@@ -1092,6 +1092,365 @@ func TestDiscard(t *testing.T) {
+		require.Nil(t, err)
+		err = networkRootMetadata.AddGlobalRule(tufv01.NewGlobalRuleThreshold(networkThresholdName, networkThresholdPatterns, networkThresholdAmount))
+		require.Nil(t, err)
+
+		// Stage both changes
+		controllerRootEnv, err = dsse.CreateEnvelope(controllerRootMetadata)
+		require.Nil(t, err)
+		controllerRootEnv, err = dsse.SignEnvelope(testCtx, controllerRootEnv, signer)
+		require.Nil(t, err)
+		controllerState.Metadata.RootEnvelope = controllerRootEnv
+		err = controllerState.preprocess()
+		require.Nil(t, err)
+		err = controllerState.Commit(controllerRepository, "Add controller repo global rule\n", true, false)
+		require.Nil(t, err)
+
+		newNetworkRootEnv, err := dsse.CreateEnvelope(networkRootMetadata)
+		require.Nil(t, err)
+		newNetworkRootEnv, err = dsse.SignEnvelope(testCtx, newNetworkRootEnv, signer)
+		require.Nil(t, err)
+		networkState.Metadata.RootEnvelope = newNetworkRootEnv
+		err = networkState.preprocess()
+		require.Nil(t, err)
+		err = networkState.Commit(networkRepository, "Add network repo global rule\n", true, false)
+		require.Nil(t, err)
+		networkStagingTip, err = networkRepository.GetReference(PolicyStagingRef)
+		require.Nil(t, err)
+
+		// 4. Apply the controller's changes and propagate
+		err = Apply(testCtx, controllerRepository, false)
+		require.Nil(t, err)
+		err = rsl.PropagateChangesFromUpstreamRepository(networkRepository, controllerRepository, networkRootMetadata.GetPropagationDirectives(), false)
+		require.Nil(t, err)
+
+		// The network repository's staging ref should not have changed since
+		// the last time, we have not staged any new changes locally
+		newNetworkStagingTip, err := networkRepository.GetReference(PolicyStagingRef)
+		require.Nil(t, err)
+		assert.Equal(t, networkStagingTip, newNetworkStagingTip)
+
+		// Load the updated repository states
+		controllerState, err = LoadCurrentState(testCtx, controllerRepository, PolicyRef)
+		require.Nil(t, err)
+		networkState, err = LoadCurrentState(testCtx, networkRepository, PolicyRef)
+		require.Nil(t, err)
+
+		// Check that controller metadata has in fact been propagated...
+		assert.Equal(t, controllerState.Metadata.RootEnvelope, networkState.ControllerMetadata["controller"].RootEnvelope)
+		// ...and that other metadata has remained the same.
+		assert.Equal(t, networkRootEnv, networkState.Metadata.RootEnvelope)
+		assert.Nil(t, networkState.Metadata.TargetsEnvelope)
+
+		// These should not be equal, as policy has been updated, but *not*
+		// policy-staging
+		networkPolicyTip, err = networkRepository.GetReference(PolicyRef)
+		require.Nil(t, err)
+		networkStagingTip, err = networkRepository.GetReference(PolicyStagingRef)
+		require.Nil(t, err)
+		assert.NotEqual(t, networkPolicyTip, networkStagingTip)
+
+		// Now, run the reconciliation workflow
+		err = ReconcileStaging(networkRepository, false)
+		assert.Nil(t, err)
+		err = Apply(testCtx, networkRepository, false)
```

Comment:
Apply calls reconcile too...

I'd do reconcile, compare tips again, then apply if we really want policy and staging synced.